### PR TITLE
add useRoutes example

### PR DESCRIPTION
**Pull request by mcansh**
closes REM-360

Signed-off-by: Logan McAnsh <logan@mcan.sh>

### CODE REVIEW

File: examples/useroutes/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# useRoutes Example
+
+This example demonstrates how to use the useRoutes hook to define routes.
+
+## Preview
+
+Open this example on [StackBlitz](https://stackblitz.com):
+
+[![Open in StackBlitz](https://developer.stackblitz.com/img/open_in_stackblitz.svg)](https://stackblitz.com/github/remix-run/react-router/tree/dev/examples/useroutes)
```
Comment (mjackson) [inline examples/useroutes/README.md:9]: Same thing about using `?file` param

**Review by mjackson — CHANGES_REQUESTED** [review]
https://stackblitz.com/edit/github-tpcbmd?file=src%2FApp.tsx

**Comment by mjackson** [comment]
Just a reminder to move `BrowserRouter` into `main.tsx` in all the examples.

Also, get the resolve stuff out of tsconfig.json in all examples

File: examples/basic/src/App.tsx [hunk 2]
```diff
@@ -1,35 +1,27 @@
 import * as React from "react";
-import {
```
Comment (mjackson) [inline examples/basic/src/App.tsx:2]: Let's update the basic example in a separate PR.

**Review by mjackson — CHANGES_REQUESTED** [review]

File: examples/basic/src/App.tsx (hunk 2 above)
Comment (mcansh) [reply examples/basic/src/App.tsx:2]: yeah, i did, but it slipped in here too 💀, will remove!

**Pull request merged**